### PR TITLE
Update quick start guide with latest info

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Updates the quick start guide to the latest lightning version. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @rohitgr7 @Felonious-Spellfire @tchaton

### CODE REVIEW

File: docs/source-app/quickstart.rst [hunk 1]
```diff
@@ -8,27 +8,24 @@ Quick Start
...
 .. code-block:: bash
 
-    lightning install quick-start
+    lightning install app lightning/quick-start
 
 And download the training script used by the App:
 
-.. code-block:: bash
-
-    curl https://gist.githubusercontent.com/tchaton/b81c8d8ba0f4dd39a47bfa607d81d6d5/raw/8d9d70573a006d95bdcda8492e798d0771d7e61b/train_script.py > train_script.py
```
Comment (awaelchli) [inline docs/source-app/quickstart.rst:31]: This comes already with the quick start, no need to download

File: docs/source-app/quickstart.rst [hunk 2]
```diff
@@ -49,38 +46,39 @@ And that's it!
...
+        Running on local URL:  http://127.0.0.1:62782/
         ...
 
 
-The build command will launch the app admin panel UI. In your app admin, you can track your app's progress, or click on the **Open App** button to view see your app's UI:
+The app will open your browser and show an interactive demo:
+
+.. figure:: https://pl-flash-data.s3.amazonaws.com/assets_lightning/quick-start/qiuck-start-tensorboard-tab.png
+    :alt: Quick Start UI - Model Training Tab
+    :width: 100 %
```
Comment (awaelchli) [inline docs/source-app/quickstart.rst:78]: I created the new screenshots myself. Check if that's good.

File: docs/source-app/quickstart.rst [hunk 3]
```diff
@@ -116,13 +114,6 @@ Simply add ``--cloud`` to run this application in the cloud 🤯
 
     lightning run app app.py --cloud
 
-And with just one line of code, run on cloud GPUs!
-
-.. code-block:: bash
-
-    USE_GPU=1
-    lightning run app app.py --cloud
```
Comment (awaelchli) [inline docs/source-app/quickstart.rst:124]: This does not work. The quick start code does not have any code that handles this.

**Review by rohitgr7 — APPROVED** [review]

File: docs/source-app/quickstart.rst (hunk 2 above)
Comment (Felonious-Spellfire) [reply docs/source-app/quickstart.rst:78]: They look great. Nice work.

**Review by otaj — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source-app/quickstart.rst (hunk 3 above)
Comment (tchaton) [reply docs/source-app/quickstart.rst:124]: Yes, this is out-dated. It needs to be passed with --app_args

**Pull request merged**